### ISSUE TITLE
[Bug] Quantization for Llama-2-70b-hf failed 

### ISSUE THREAD

**Issue by PYNing** [InternLM/lmdeploy#249]
### Checklist

- [X] 1. I have searched related issues but cannot get the expected help.
- [X] 2. The bug has not been fixed in the latest version.

### Describe the bug

Quantization tools works fine with 7B model（Llama-2-7b-hf），but encountered a crash when quantize 70B model（Llama-2-70b-hf）

All weights were downloaded from HuggingFace without any modification.

Version: 0.5.0
Device:  4090*8

### Reproduction

python -m lmdeploy.lite.apis.calibrate --models Llama-2-70b-hf --calib_dataset 'wikitext2' --calib_samples 128 --calib_seqlen 128 --work_dir /tmp

### Error traceback

```Shell
Traceback (most recent call last):                                                                                                                                                                    
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main                                                                                                                                
    return _run_code(code, main_globals, None,                                                                                                                                                        
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code                                                                                                                                           
    exec(code, run_globals)                                                                                                                                                                           
  File "/usr/local/lib/python3.8/dist-packages/lmdeploy/lite/apis/calibrate.py", line 110, in <module>                                                                                                
    fire.Fire(calibrate)                                                                                                                                                                              
  File "/usr/local/lib/python3.8/dist-packages/fire/core.py", line 141, in Fire                                                                                                                       
    component_trace = _Fire(component, args, parsed_flag_args, context, name)                                                                                                                         
  File "/usr/local/lib/python3.8/dist-packages/fire/core.py", line 475, in _Fire                                                                                                                      
    component, remaining_args = _CallAndUpdateTrace(                                                                                                                                                  
  File "/usr/local/lib/python3.8/dist-packages/fire/core.py", line 691, in _CallAndUpdateTrace                                                                                                        
    component = fn(*varargs, **kwargs)                                                                                                                                                                
  File "/usr/local/lib/python3.8/dist-packages/lmdeploy/lite/apis/calibrate.py", line 101, in calibrate                                                                                               
    calib_ctx.calibrate(all_data)                                                                                                                                                                     
  File "/usr/local/lib/python3.8/dist-packages/lmdeploy/lite/quantization/calibration.py", line 274, in calibrate                                                                                     
    _ = self.model.model(data.to(self.device))                                                                                                                                                        
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py", line 1194, in _call_impl                                                                                                  
    return forward_call(*input, **kwargs)                                                                                                                                                             
  File "/usr/local/lib/python3.8/dist-packages/transformers/models/llama/modeling_llama.py", line 693, in forward                                                                                     
    layer_outputs = decoder_layer(                                                                                                                                                                    
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py", line 1194, in _call_impl                                                                                                  
    return forward_call(*input, **kwargs)                                                                                                                                                             
  File "/usr/local/lib/python3.8/dist-packages/lmdeploy/lite/quantization/calibration.py", line 164, in _forward                                                                                      
    k_obs.observe(key)                                                                                                                                                                                
  File "/usr/local/lib/python3.8/dist-packages/torch/autograd/grad_mode.py", line 27, in decorate_context                                                                                             
    return func(*args, **kwargs)                                                                                                                                                                      
  File "/usr/local/lib/python3.8/dist-packages/lmdeploy/lite/quantization/activation/observer.py", line 41, in observe                                                                                
    assert x.size(2) == self.num_head                                                                                                                                                                 
AssertionError
```

**Comment by pppppM** [comment]
Llama2 70B uses Group Query Attention, which was not considered during the development of the quantization feature. We will fix this as soon as possible. To avoid impacting your use, you can temporarily comment out this line https://github.com/InternLM/lmdeploy/blob/f06db80debeb420e0ff556ef3248bed06763d325/lmdeploy/lite/quantization/activation/observer.py#L41.

**Comment by pppppM** [comment]
@PYNing 
What new errors raised? After commenting this line, I was able to successfully export the 70B int4 weights.

**Comment by PYNing** [comment]
@pppppM 

Terrible sorry for late reply, the new error has been pasted below.
Has this PR https://github.com/InternLM/lmdeploy/pull/262 fixed it?

```
root@4fd8f2bb4888:/ttt# python3 -m lmdeploy.lite.apis.calibrate --model Llama-2-70b-hf --calib_dataset 'wikitext2' --calib_samples 128 --calib_seqlen 128 --work_dir ./tmp
Loading checkpoint shards: 100%|███████████████████████████████████████████████████████████████████████████████████████████| 15/15 [01:10<00:00,  4.68s/it]
Loading calibrate dataset ...
Using the latest cached version of the module from /root/.cache/huggingface/modules/datasets_modules/datasets/wikitext/a241db52902eaf2c6aa732210bead40c090019a499ceb13bcbfa3f8ab646a126 (last modified on Tue Aug 15 12:19:53 2023) since it couldn't be found locally at wikitext., or remotely on the Hugging Face Hub.
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.8/dist-packages/lmdeploy/lite/apis/calibrate.py", line 110, in <module>
    fire.Fire(calibrate)
  File "/usr/local/lib/python3.8/dist-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/usr/local/lib/python3.8/dist-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/usr/local/lib/python3.8/dist-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/lmdeploy/lite/apis/calibrate.py", line 101, in calibrate
    calib_ctx.calibrate(all_data)
  File "/usr/local/lib/python3.8/dist-packages/lmdeploy/lite/quantization/calibration.py", line 274, in calibrate
    _ = self.model.model(data.to(self.device))
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/transformers/models/llama/modeling_llama.py", line 693, in forward
    layer_outputs = decoder_layer(
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/lmdeploy/lite/quantization/calibration.py", line 164, in _forward
    k_obs.observe(key)
  File "/usr/local/lib/python3.8/dist-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/lmdeploy/lite/quantization/activation/observer.py", line 48, in observe
    self.max_val = torch.maximum(self.max_val, cur_max)
RuntimeError: The size of tensor a (64) must match the size of tensor b (8) at non-singleton dimension 0
```

**Comment by pppppM** [comment]
#262 don't fix the issue.
I'll fix this issue today @PYNing

**Comment by pppppM** [comment]
@PYNing Llama2 70 has been fixed in #273

**Comment by PYNing** [comment]
> @PYNing Llama2 70 has been fixed in #273

Thank you so much for your hard work, I will use this PR in advance.
